ui/lib/kv/addon/components/kv-list-filter.js

```diff
@@ -13,42 +13,59 @@ import { tracked } from '@glimmer/tracking';
 
 /**
  * @module KvListFilter
- * `KvListFilter` component is used to filter through the KV metadata LIST response. It allows users to search through the current list, navigate into directories, and use keyboard functions to: autocomplete, view a secret, create a new secret, or clear the input field.
- *
- * @param {object} model - The adapter model object which contains an array of secret models.
+ * `KvListFilter` filters through the KV metadata LIST response. It allows users to search through the current list, navigate into directories, and use keyboard functions to: autocomplete, view a secret, create a new secret, or clear the input field.
+ *  *
+ * <KvListFilter
+ *  @secrets={{this.model.secrets}}
+ *  @mountPoint={{this.model.mountPoint}}
+ *  @filterValue="beep/my-"
+ *  @pageFilter="my-"
+ * />
+ * @param {array} secrets - An array of secret models.
  * @param {string} mountPoint - Where in the router files we're located. For this component it will always be vault.cluster.secrets.backend.kv
  * @param {string} filterValue - A concatenation between the list-directory's dynamic path "path-to-secret" and the queryParam "pageFilter". For example, if we're inside the beep/ directory searching for any secret that starts with "my-" this value will equal "beep/my-".
- * @param {string} pageFilter - The queryParam value.
+ * @param {string} pageFilter - The queryParam value, does not include pathToSecret ex: my-.
  */
 
 export default class KvListFilterComponent extends Component {
   @service router;
   @tracked filterIsFocused = false;
 
-  kvRoute(route) {
-    return `${this.args.mountPoint}.${route}`;
+  navigate(pathToSecret, pageFilter) {
+    const route = pathToSecret ? `${this.args.mountPoint}.list-directory` : `${this.args.mountPoint}.list`;
+    const args = [route];
+    if (pathToSecret) {
+      args.push(pathToSecret);
+    }
+    args.push({
+      queryParams: {
+        pageFilter: pageFilter ? pageFilter : null,
+      },
+    });
+    this.router.transitionTo(...args);
```

Comment:
I think this means we'll hit the `kv/config` endpoint each time we call this method

Reply:
It should always transition to either `vault.secrets.backend.kv.list` or `vault.secrets.backend.kv.list-directory`. This is because of the route returned on row 35. here is a screenshot of the args passed to the transitionTo
![image](https://github.com/hashicorp/vault/assets/6618863/a0bf9ba5-789e-470b-9c44-bc8251339067)

Reply:
I see that. We were transitioning before (now it's just all handled in a method). And this is the same flow as we currently have in KV. I'm not sure a way around this. Is this something to note, or something you think we need to address?

Reply:
A user had commented on it being problematic (maybe just annoying) when this all 403s - so I think it was a request for us to minimize requests to this endpoint.

I think it's something we should address for this release, but not necessarily in this PR